### PR TITLE
feat: disable Standard field edition in Relation field form

### LINKED ISSUE #2584: Object Field Edit - Disable standard field edition in Relation preview form
## Context

See #2340 

Currently, on "Object Field Edit" page: in the "Relation" field type preview form, the related field's icon and label are always editable.

For this issue: If the related field is "Standard" (`isCustom` is `false`), disable editing the related field's icon and label.

Example: in the image below, the "Field on People" icon picker and input should be disabled if "Working in" is a "Standard" field.

## Figma

Image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=14407-174415&mode=design&t=nJ2HHxP3xoTZJB7T-11

### Scenario: Creating a "Relation" field

**GIVEN** I am on a Object Field Edit page of a "Relation" field
**AND** the related field is a "Standard" field
**THEN** I cannot edit the related field's icon and label (form is disabled)

### PR DESCRIPTION
**Pull request by thaisguigon**
Closes #2584

### CODE REVIEW

**Review by charlesBochet — APPROVED** [review]
LGTM

File: front/src/modules/settings/data-model/components/SettingsObjectFieldRelationForm.tsx [hunk 1]
```diff
@@ -18,6 +18,7 @@ export type SettingsObjectFieldRelationFormValues = Partial<{
 }>;
 
 type SettingsObjectFieldRelationFormProps = {
+  disableFieldEdition?: boolean;
```
Comment (charlesBochet) [inline front/src/modules/settings/data-model/components/SettingsObjectFieldRelationForm.tsx:21]: disableFieldNamesEdition?

File: front/src/modules/settings/data-model/components/SettingsObjectFieldRelationForm.tsx [hunk 2]
```diff
@@ -18,6 +18,7 @@ export type SettingsObjectFieldRelationFormValues = Partial<{
...
   disableRelationEdition?: boolean;
```
Comment (charlesBochet) [inline front/src/modules/settings/data-model/components/SettingsObjectFieldRelationForm.tsx:22]: disableRelationObjectEdition?

File: front/src/modules/settings/data-model/components/SettingsObjectFieldTypeSelectSection.tsx [hunk 3]
```diff
@@ -127,7 +127,10 @@ export const SettingsObjectFieldTypeSelectSection = ({
             form={
               values.type === FieldMetadataType.Relation && (
                 <SettingsObjectFieldRelationForm
-                  disableRelationEdition={!!relationFieldMetadataId}
+                  disableFieldEdition={
+                    !!relationFieldMetadata && !relationFieldMetadata.isCustom
+                  }
+                  disableRelationEdition={!!relationFieldMetadata?.id}
```
Comment (charlesBochet) [inline front/src/modules/settings/data-model/components/SettingsObjectFieldTypeSelectSection.tsx:133]: would something like isNotEmpty(relationFieldMetadata) work? I'm a bit confused by the fact we check the id specifically

File: front/src/modules/settings/data-model/components/SettingsObjectFieldTypeSelectSection.tsx [hunk 4]
```diff
@@ -127,7 +127,10 @@ export const SettingsObjectFieldTypeSelectSection = ({
             form={
               values.type === FieldMetadataType.Relation && (
                 <SettingsObjectFieldRelationForm
-                  disableRelationEdition={!!relationFieldMetadataId}
+                  disableFieldEdition={
+                    !!relationFieldMetadata && !relationFieldMetadata.isCustom
```
Comment (charlesBochet) [inline front/src/modules/settings/data-model/components/SettingsObjectFieldTypeSelectSection.tsx:131]: relationFieldMetadata && !relationFieldMetadata.isCustom

**Review by charlesBochet — APPROVED** [review]

File: front/src/modules/settings/data-model/components/SettingsObjectFieldRelationForm.tsx (hunk 1 above)
Comment (thaisguigon) [reply front/src/modules/settings/data-model/components/SettingsObjectFieldRelationForm.tsx:21]: This disables the icon picker (for the field icon) and the input (for the field name), so it does not disable only the field name edition!

File: front/src/modules/settings/data-model/components/SettingsObjectFieldRelationForm.tsx (hunk 2 above)
Comment (thaisguigon) [reply front/src/modules/settings/data-model/components/SettingsObjectFieldRelationForm.tsx:22]: This disables not only selecting the relation object, but also the relation type (Has many/Has one/Belongs to one).

File: front/src/modules/settings/data-model/components/SettingsObjectFieldTypeSelectSection.tsx (hunk 4 above)
Comment (thaisguigon) [reply front/src/modules/settings/data-model/components/SettingsObjectFieldTypeSelectSection.tsx]: Done

**Pull request merged**